src/main/java/gnosis/task/TaskStorageManager.java

```diff
@@ -0,0 +1,104 @@
+package gnosis.task;
+
+import gnosis.model.*;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TaskStorageManager {
+
+    private static final String DIRECTORY_PATH = "data";
+    private static final String FILE_PATH = DIRECTORY_PATH + "/tasks.csv";
+    private static final File file = new File(FILE_PATH);
+    private static final String  DELIMITER = ",";
+
+
+    public static List<Task> loadGnosisTasks() {
+        //check if user has folder:
+        // if no folder -> means no data found -> create one from scratch
+        // if folder found -> load to arraylist tasks
+        if (TaskStorageManager.isDataFileAvail()) {
+            return TaskStorageManager.getTasksFromFile();
+        } else {
+            TaskStorageManager.createDataFolder();
+            return new ArrayList<>();
+        }
+    }
+
+    public static List<Task> getTasksFromFile() {
+        List<Task> tasks = new ArrayList<>();
+        try {
+            tasks = Files.newBufferedReader(Paths.get(FILE_PATH))
+                    .lines()
+                    .skip(1)
+                    .map(s -> {
```

Comment:
Any possible reason why your `map` method has a nested function block in it? Maybe to make your code neater, you can try and push the function block out as a private method, if its not state-dependent. Example:
``` java
public static List<Task> getTasksFromFile() {
    List<Task> tasks = new ArrayList<>();
    try {
        tasks = Files.newBufferedReader(Paths.get(FILE_PATH))
                .lines()
                .skip(1)
                .map(this::parseTask)
                .collect(Collectors.toList());
    } catch (IOException e) {
        e.printStackTrace();
    }
    return tasks;
}

private Task parseTask(String s) throws IOException {
    // migrated method contents omitted
}
```